tests/test_io.py

```diff
@@ -7,7 +8,115 @@
 from s3_upload.utils import io
 
 
-@patch("s3_upload.utils.io.listdir")
+class TestAcquireLock(unittest.TestCase):
+    test_lock = os.path.join(TEST_DATA_DIR, "test.lock")
+
+    def setUp(self):
+        if os.path.exists(self.test_lock):
+            os.remove(self.test_lock)
+
+    def tearDown(self):
+        if os.path.exists(self.test_lock):
+            os.remove(self.test_lock)
+
+    @patch("s3_upload.utils.io.os.open", wraps=os.open)
+    def test_when_lock_file_does_not_exist_the_correct_flag_set(
+        self, mock_open
+    ):
+        """
+        When the lock file does not already exist we should open the lock
+        file with O_RDWR (2), O_CREAT (64) and O_TRUNC (512). Therefore
+        the flag set for open should be 578
+        """
+        io.acquire_lock(self.test_lock)
+
+        self.assertEqual(mock_open.call_args[1]["flags"], 578)
+
+    @patch("s3_upload.utils.io.os.open", wraps=os.open)
+    def test_when_lock_file_exists_the_correct_flag_set(self, mock_open):
+        """
+        When the lock file already exists we should open the lock file
+        with only O_RDWR (2) and therefore the flag should be 2, which
+        will prevent the file from being truncated and losing contents.
+        """
+        open(self.test_lock, "w").close()
+
+        io.acquire_lock(self.test_lock)
+
+        self.assertEqual(mock_open.call_args[1]["flags"], 2)
+
+    def test_file_lock_correctly_acquired_when_not_already_set(self):
+        """
+        If we can successfully acquire a file lock we should get an integer
+        file descriptor returned and the expected message written to the file
+        """
+        lock_fd = io.acquire_lock(self.test_lock)
+
+        with open(self.test_lock) as fh:
+            lock_contents = fh.read()
+
+        with self.subTest("file descriptor returned"):
+            self.assertEqual(type(lock_fd), int)
+
+        with self.subTest("lock contents correct"):
+            expected_contents = (
+                r"file lock acquired from running upload at"
+                r" [\d]{2}:[\d]{2}:[\d]{2} from process"
+                r" [\d]+"
+            )
+            self.assertRegex(lock_contents, expected_contents)
+
+    def test_correctly_exit_when_lock_already_present(self):
```

Comment:
Would you also expect the print statement ("Colud not acquire exclusive lock ...") here?